docs/DeveloperGuide.md

```diff
@@ -153,6 +153,137 @@ The following sequence diagram shows how the undo operation works:
+
+#### Design Considerations
+##### Aspect 1: Concern while adding a new feature
+* Workflow must be consistent with other searching commands e.g. search recipe.
+
+##### Aspect 2: How do we successfully search and filter the recipes based on the user' search 
+* **Alternative 1 (current choice):** User can only search for recipes based on one fields at a time
+  * Pros: Easy to implement.
+  * Cons: User's cannot filter the recipes by two or three fields at once
+
+* **Alternative 2:** User can only search for recipes by all fields at once
+  * Pros: Harder to implement.
+  * Cons: User's can filter the recipes by two or three fields at once
+
+### Delete Consumption feature
+
+#### Implementation
+This feature allows users to delete the recipes they have eaten in the calorie tracker.
+
+Substitutability is used in Command and Parser:
+* `DeleteConsumptionCommand` extends `Command`
+* `DeleteConsumptionCommandParser` implements `Parser<DeleteConsumptionCommand>`
+
+Given below is an example usage scenario and how the mechanism behaves at each step.
+
+![DeleteConsumptionSequence](images/DeleteConsumptionSequence.png)
+
+Step 1:
+User inputs the delete consumption command to delete the recipes eaten in the calorie tracker.
+
+Step 2:
+After successful parsing of user input, the `DeleteConsumptionCommand#execute(Model model)` method is called.
+
+Step 3:
+The recipe that the user has specified will be deleted from the consumption list.
+
+Step 4:
+After the successful deleting of recipes, a `CommandResult` object is instantiated and returned to `LogicManager`.
+
+#### Design Considerations
+##### Aspect 1: Concern while adding a new feature
+* Workflow must be consistent with other deleting commands e.g. delete recipe and delete ingredient.
+
+### Recommend feature
+
+#### Implementation
+This feature allows users to get recommended recipes that they are able to make with the ingredients in their fridge.
+
+Substitutability is used in Command:
+* `RecommendCommand` extends `Command`
+
+Given below is an example usage scenario and how the mechanism behaves at each step.
+
+![RecommendSequence](images/RecommendSequence.png)
+
+Step 1:
+User inputs the recommend command to get the recommended recipes.
+
+Step 2:
+After successful parsing of user input, the `RecommendCommand#execute(Model model)` method is called.
+
+Step 3:
+The list of recommended recipes will be returned to the user.
+
+Step 4:
+After the successful recommending of recipes, a `CommandResult` object is instantiated and returned to `LogicManager`.
+
+#### Design Considerations
+##### Aspect : How do we quickly and accurately compare ingredients in each recipe and the user's fridge
+* **Alternative 1 (current choice):** Compare the exact ingredients in each recipe to the users ingredients in the fridge
+  * Pros: Easy to implement
+  * Cons: Slow to compare, the ingredients might not match if the spellings are different
```

Comment:
maybe another con could be other than spelling diff we could say diff ingredient names but similar ingredients in real life eg mozarella and cheese. also another con is if user dont have the basic ingredients in fridge eg salt, pepper, soya sauce, it may not be recommended to user